streams/src/main/java/org/apache/kafka/streams/processor/DefaultPartitionGrouper.java

```diff
@@ -61,7 +61,7 @@ protected int maxNumPartitions(Cluster metadata, Set<String> topics) {
             List<PartitionInfo> infos = metadata.partitionsForTopic(topic);
 
             if (infos == null)
-                throw new KafkaException("topic not found :" + topic);
+                throw new KafkaException("Topic not found during partition assignment: " + topic);
```

Comment:
Currently the entry class `KafkaStreams` does not throw any exceptions since all exceptions are thrown in its spawned `StreamThread`, which cannot by caught via user code and upon happening will just cause this thread to die and other threads taking over the tasks. 

We may need to "hook up" the exceptions thrown from StreamThreads to KafkaStreams so that users can actively catch and act upon those.